jupyterlab_gitsync/src/components/toolbar_buttons.tsx

```diff
@@ -0,0 +1,169 @@
+import * as React from 'react';
+import { classes, style } from 'typestyle';
+import {
+  toolbarItemClass,
+  toolbarButtonClass,
+  toolbarButtonIconClass,
+} from '../style/toolbar';
+
+import { Props } from './panel';
+
+export interface State {
+  state: string;
+  label: string;
+  style: ReturnType<typeof style>;
+}
+
+class ToolbarButton extends React.Component<Props, State> {
```

Comment:
You may want to make this class abstract since I don't think you can directly use this with the normal React TSX syntax due to its constructor requiring additional parameters beyond its props.

Reply:
Will do! I'll make the changes in a future PR.